File: leetcode-solutions/1236-n-th-tribonacci-number/solution.py
class Solution(object):
    def tribonacci(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0:
            return 0
        if n < 3:
            return 1
        prev, mid, next = 0, 1, 1
        for i in range (3, n + 1):
            curr = prev + mid + next
            prev = mid
            mid = next
            next = curr
        return next

File: leetcode-solutions/1236-n-th-tribonacci-number/solution.java
class Solution {
    public int tribonacci(int n) {
        int prev = 0, mid = 1, next = 1;
        if(n == 0)
            return prev;
        if(n < 3)
            return 1;
        for(int i = 3;i < n + 1;i++) {
            int curr = prev + mid + next;
            prev = mid;
            mid = next;
            next = curr;
        }
        return next;
    }
}
